File: bench/sumcol/sumcol.php
#!/usr/local/bin/php -f
<?php
/*
 $Id: sumcol.php,v 1.1 2004-05-29 08:48:06 bfulgham Exp $
 http://www.bagley.org/~doug/shootout/
*/
$fd = fopen("php://stdin", "r");
$sum = 0;
while (!feof ($fd)) { $sum += fgets($fd, 1024); }
fclose($fd);
print "$sum\n";
?>

File: bench/sumcol/sumcol.lua
-- $Id: sumcol.lua,v 1.2 2004-06-12 16:19:44 bfulgham Exp $
-- http://www.bagley.org/~doug/shootout/

local sum = 0
local line = io.read()

while line ~= nil do
    sum = sum + line
    line = io.read()
end
io.write(sum, "\n")
